### PR TITLE
py domain: Use AST parser to convert signature to doctree

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature

### Purpose
- As a preparation of #6990, Use AST parser to convert signature to doctree
- It allows to decorate each tokens in latter step.
- Originally, this is a part of #6990.

### CODE REVIEW

File: tests/test_domain_py.py [hunk 1]
```diff
@@ -241,7 +242,63 @@ def test_pyfunction_signature(app):
...
+                                                        "=None")])])
+
+    # case: separator in the middle
+    text = ".. py:function:: hello(a, /, b, *, c)"
+    doctree = restructuredtext.parse(app, text)
+    assert_node(doctree[1][0][1],
+                [desc_parameterlist, ([desc_parameter, "a"],
+                                      "/",
+                                      [desc_parameter, "b"],
+                                      "*",
```
Comment (tk0miya) [inline tests/test_domain_py.py:292]: So far, all children of `desc_parameter` are only `desc_parameter`. But this PR makes `Text` node as children of it. I don't think this is a breaking change, but...

Comment (tk0miya) [reply tests/test_domain_py.py:292]: Note: The `Text` nodes will be replaced by `inline` node in #6990.

Comment (shimizukawa) [reply tests/test_domain_py.py:292]: I think it's OK.

Comment (tk0miya) [reply tests/test_domain_py.py:292]: Thank you for comment!

**Pull request merged**